File: src/main/java/com/bdev/hogwarts_api/rest_service/event_participants/EventParticipantsRestService.java
package com.bdev.hogwarts_api.rest_service.event_participants;

import acropollis.municipali.security.common.dto.MunicipaliUserInfo;
import com.bdev.hogwarts_api.data.dto.events.EventParticipant;

import java.util.List;

public interface EventParticipantsRestService {
    List<EventParticipant> getAllParticipants(MunicipaliUserInfo userInfo);
    EventParticipant getEventParticipant(MunicipaliUserInfo userInfo, long eventParticipantId);
    List<EventParticipant> getEventParticipants(MunicipaliUserInfo userInfo, long eventId);
    long createEventParticipant(MunicipaliUserInfo userInfo, EventParticipant eventParticipant);
    void editEventParticipant(MunicipaliUserInfo userInfo, EventParticipant eventParticipant);
    void deleteEventParticipant(MunicipaliUserInfo userInfo, long eventParticipantId);
}

File: src/main/java/com/bdev/hogwarts_api/rest_service/event_participants/EventParticipantsRestService.kt
package com.bdev.hogwarts_api.rest_service.event_participants

import acropollis.municipali.security.common.dto.MunicipaliUserInfo
import com.bdev.hogwarts_api.data.dto.events.EventParticipant

interface EventParticipantsRestService {
    fun getAllParticipants(userInfo: MunicipaliUserInfo): List<EventParticipant>
    fun getEventParticipant(userInfo: MunicipaliUserInfo, eventParticipantId: Long): EventParticipant
    fun getEventParticipants(userInfo: MunicipaliUserInfo, eventId: Long): List<EventParticipant>
    fun createEventParticipant(userInfo: MunicipaliUserInfo, eventParticipant: EventParticipant): Long
    fun editEventParticipant(userInfo: MunicipaliUserInfo, eventParticipant: EventParticipant)
    fun deleteEventParticipant(userInfo: MunicipaliUserInfo, eventParticipantId: Long)
}
